extensions/resteasy-classic/rest-client/config/src/main/java/io/quarkus/restclient/config/RestClientConfigRoot.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.restclient.config;
+
+import java.util.Map;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "rest-client", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class RestClientConfigRoot {
```

Comment:
`RestClientConfig` seems more suitable as a name for a public API if we expect users to inject it directly into their services. Usually extension configuration roots are not used outside extensions so the name doesn't matter much. In this case, we may want to be careful about that.

Reply:
I'm going to change the root to `RestClientsConfig`, while the nested config class would be `RestClientConfig` (for specific client configurations), similarly as GraphQL extension does (https://github.com/quarkusio/quarkus/pull/17973/files#diff-b292b26a4dd3fd549d35c404b15f61fc9e8923639e256774ab134a5f5797ba95).